src/components/InitialURLContextProvider.tsx

```diff
@@ -21,46 +18,31 @@ const InitialURLContext = createContext<InitialUrlContextType>({
     setInitialURL: () => {},
 });
 
-type InitialURLContextProviderProps = {
-    /** URL passed to our top-level React Native component by HybridApp. Will always be undefined in "pure" NewDot builds. */
-    url?: Route | ValueOf<typeof CONST.HYBRID_APP>;
-
+type InitialURLContextProviderProps = AppProps & {
     /** Children passed to the context provider */
     children: ReactNode;
 };
 
-function InitialURLContextProvider({children, url}: InitialURLContextProviderProps) {
+function InitialURLContextProvider({children, url, hybridAppSettings, timestamp}: InitialURLContextProviderProps) {
     const [initialURL, setInitialURL] = useState<Route | undefined>();
-    const [lastVisitedPath] = useOnyx(ONYXKEYS.LAST_VISITED_PATH);
     const {splashScreenState, setSplashScreenState} = useSplashScreenStateContext();
 
     useEffect(() => {
-        if (url !== CONST.HYBRID_APP.REORDERING_REACT_NATIVE_ACTIVITY_TO_FRONT) {
-            return;
-        }
-
-        if (splashScreenState !== CONST.BOOT_SPLASH_STATE.HIDDEN) {
-            setSplashScreenState(CONST.BOOT_SPLASH_STATE.READY_TO_BE_HIDDEN);
-            Navigation.navigate(lastVisitedPath as Route);
-        }
-    }, [lastVisitedPath, setSplashScreenState, splashScreenState, url]);
-
-    useEffect(() => {
-        if (url === CONST.HYBRID_APP.REORDERING_REACT_NATIVE_ACTIVITY_TO_FRONT) {
-            return;
-        }
-
-        if (url) {
-            signInAfterTransitionFromOldDot(url).then((route) => {
-                setInitialURL(route);
+        if (url && hybridAppSettings) {
+            setInitialURL(url);
+            signInAfterTransitionFromOldDot(hybridAppSettings).then(() => {
+                if (splashScreenState === CONST.BOOT_SPLASH_STATE.HIDDEN) {
+                    return;
+                }
                 setSplashScreenState(CONST.BOOT_SPLASH_STATE.READY_TO_BE_HIDDEN);
             });
             return;
         }
         Linking.getInitialURL().then((initURL) => {
```

Comment:
I'm curious about the relevance of this line. Do we even need it?

Reply:
Ah, it’s probably still needed for the standalone NewDot. Even though HybridApp is the only one available on the App Store and Play Store, we still need this for external contributors 🤔

Reply:
Probably true but this is old code so I'm not sure